R/helper-quarto.R

```diff
@@ -16,44 +17,51 @@ quarto_multilingual_book <- function(parent_dir,
                                      main_language = "en",
                                      further_languages = c("es", "fr"),
                                      register_languages = TRUE,
-                                     site_url = "https://example.com") {
+                                     site_url = "https://example.com",
+                                     placement = c("sidebar", "navbar")) {
   quarto_multilingual_project(
     parent_dir = parent_dir,
     project_dir = project_dir,
     type = "book",
     main_language = main_language,
     further_languages = further_languages,
     register_languages = register_languages,
-    site_url = site_url
+    site_url = site_url,
+    placement = placement
   )
 }
 
 #' @rdname quarto_multilingual_book
 #' @export
 quarto_multilingual_website <- function(parent_dir,
-                                       project_dir,
-                                       main_language = "en",
-                                       further_languages = c("es", "fr"),
-                                       register_languages = TRUE,
-                                       site_url = "https://example.com") {
+                                        project_dir,
+                                        main_language = "en",
+                                        further_languages = c("es", "fr"),
+                                        register_languages = TRUE,
+                                        site_url = "https://example.com",
+                                        placement = c("navbar", "sidebar")) {
   quarto_multilingual_project(
     parent_dir = parent_dir,
     project_dir = project_dir,
     type = "website",
     main_language = main_language,
     further_languages = further_languages,
     register_languages = register_languages,
-    site_url = site_url
+    site_url = site_url,
+    placement = placement
   )
 }
 
 quarto_multilingual_project <- function(parent_dir,
-                                       project_dir,
-                                       type = c("book", "website"),
-                                       main_language = "en",
-                                       further_languages = c("es", "fr"),
-                                       register_languages = TRUE,
-                                       site_url = "https://example.com") {
+                                        project_dir,
+                                        type = c("book", "website"),
+                                        main_language = "en",
+                                        further_languages = c("es", "fr"),
+                                        register_languages = TRUE,
+                                        site_url = "https://example.com",
+                                        placement = c("navbar", "sidebar")) {
+  
+  placement <- rlang::arg_match(placement)
```

Comment:
oh wait, this needs the values (even if that repeats the function definition), at least I think so.

Reply:
Are you sure?

My test gets me exactly the result I'd expect:

```r
f <- function(x = c("1", "2")) {
  x <- rlang::arg_match(x)

  print(x)
}

f()
f("1")
f("2")
f("3")


# > f()
[1] "1"
# > f("1")
[1] "1"
# > f("2")
[1] "2"
# > f("3")
Run `rlang::last_trace()` to see where the error occurred.
Error in `f()`:
! `x` must be one of "1" or "2", not "3".
Show Traceback
```

Reply:
oooh then perfect! thank you!!